File: client/src/types/product.ts
export interface ProductCreate {
  name: string
  sku: string
  price: number
  stock_quantity: number
}

export interface ProductUpdate {
  name?: string
  sku?: string
  price?: number
  stock_quantity?: number
}

export interface ProductResponse extends ProductCreate {
  id: number
  created_at: string
  updated_at: string
}

File: server/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional
from decimal import Decimal

class ProductCreate(BaseModel):
    SKU: str
    prod_name: str
    prod_category: str
    prod_description: Optional[str] = None
    unit_price: Decimal
    inventory: int = 0

class ProductUpdate(BaseModel):
    prod_name: Optional[str] = None
    prod_category: Optional[str] = None
    prod_description: Optional[str] = None
    unit_price: Optional[Decimal] = None

class ProductInventoryUpdate(BaseModel):
    inventory: int  # New inventory quantity

class ProductResponse(BaseModel):
    prod_id: int
    store_id: int
    SKU: str
    prod_name: str
    prod_category: str
    prod_description: Optional[str]
    unit_price: Decimal
    inventory: int

    class Config:
        from_attributes = True
